apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx

```diff
@@ -53,168 +54,210 @@ const prettyTimestamp = (timestamp: number) => {
   return `${day} ${month} ${year}`;
 };
 
-export default async function Proposal({
+export default function Proposal({
   params: { proposalId, poolId, chain, garden },
 }: {
   params: { proposalId: string; poolId: number; chain: number; garden: string };
 }) {
   // TODO: fetch garden decimals in query
-  const { data: getProposalQuery } = await queryByChain<getProposalDataQuery>(
-    urqlClient,
-    chain,
-    getProposalDataDocument,
-    {
-      garden: garden,
-      proposalId: proposalId,
-    },
-  );
+  const { data: getProposalQuery } =
+    useSubgraphQueryByChain<getProposalDataQuery>(
+      chain,
+      getProposalDataDocument,
+      {
+        garden: garden,
+        proposalId: proposalId,
+      },
+      {},
+      { topic: "proposal", id: proposalId, type: "update", chainId: chain },
+    );
 
   const proposalData = getProposalQuery?.cvproposal;
 
-  if (!proposalData) {
-    return (
-      <p className="text-center text-2xl text-error">{`Proposal ${proposalId} not found`}</p>
-    );
-  }
-
   const tokenSymbol = getProposalQuery?.tokenGarden?.symbol;
   const tokenDecimals = getProposalQuery?.tokenGarden?.decimals;
-  const proposalIdNumber = proposalData.proposalNumber as number;
-  const convictionLast = proposalData.convictionLast as string;
-  const threshold = proposalData.threshold as bigint;
-  const proposalType = proposalData.strategy.config?.proposalType as number;
-  const requestedAmount = proposalData.requestedAmount as bigint;
-  const beneficiary = proposalData.beneficiary as Address;
-  const submitter = proposalData.submitter as Address;
-  const status = proposalData.proposalStatus as number;
-  const stakedAmount = proposalData.stakedAmount as bigint;
-  const metadata = proposalData.metadata;
+  const proposalIdNumber = proposalData?.proposalNumber as number;
+  const convictionLast = proposalData?.convictionLast as string;
+  const threshold = proposalData?.threshold as bigint;
+  const proposalType = proposalData?.strategy.config?.proposalType as number;
+  const requestedAmount = proposalData?.requestedAmount as bigint;
+  const beneficiary = proposalData?.beneficiary as Address;
+  const submitter = proposalData?.submitter as Address;
+  const status = proposalData?.proposalStatus as number;
+  const stakedAmount = proposalData?.stakedAmount as bigint;
+  const metadata = proposalData?.metadata;
 
   const isSignalingType = proposalType == 0;
```

Comment:
here its one example: `undefined != 0 `

Reply:
Yeah but the var is only used when its not undefined if you check the html after

Reply:
not true, its used even been undefined

Reply:
i meant that every place that I'm using them I have a null check on proposalData

Reply:
Ill also switch all to a single object

Reply:
Its much more better design, it just change a lot of read contract

Reply:
I came up with something more like before:
https://github.com/1Hive/gardens-v2/pull/229/commits/7de7a961e5d2a1ad87b6119713f6e277cab5ef5c